COMPATIBILITY.md

```diff
@@ -0,0 +1,53 @@
+# Compatibility
+
+## Default `otel` Global Providers Compatibility
+
+Auto-instrumentation can be configured to capture the telemetry sent to the
+[`otel`] default global providers.
+
+Supported versions of [`otel`]:
+
+- `v0.14.0` to `v1.28.0`
+
+[`otel`]: https://pkg.go.dev/go.opentelemetry.io/otel
+
+## Instrumented Library Compatibility
+
+Instrumentation is provided for the following Go libraries.
+
+- [`database/sql`](#database/sql)
+- [`github.com/segmentio/kafka-go`](#github.com/segmentio/kafka-go)
+- [`google.golang.org/grpc`](#google.golang.org/grpc)
+- [`net/http`](#net/http)
```

Comment:
Should we mention Gorilla Mux, since it's also working and it's a popular library?

Reply:
We don't do any testing on what versions we support as far as I know.

I think we could add that, any maybe gin as well, as a follow up to this PR. We would need to develop a way to check what versions were actually supported though.

Reply:
Captured in https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/944